File: app/src/types/ProjectTemplate.ts
import { ProjectStatus } from "./ProjectStatus";
import { Item } from "./Item";

export type ProjectTemplate = {
  id: number;
  name: string;
  description?: string;
  defaultStatus: ProjectStatus;
  createdAt: string;
  updatedAt?: string;
  items?: ProjectTemplateItem[];
};

export type ProjectTemplateItem = {
  id: number;
  templateId: number;
  itemId: string;
  quantityUsed: number;
  createdAt: string;
  item?: Item;
};

export type ProjectTemplateInput = {
  name: string;
  description?: string;
  defaultStatus: ProjectStatus;
  items?: ProjectTemplateItemInput[];
};

export type ProjectTemplateItemInput = {
  itemId: string;
  quantityUsed: number;
};

export type ProjectTemplatePage = {
  content: ProjectTemplate[];
  pageInfo: {
    totalElements: number;
    totalPages: number;
    currentPage: number;
    hasNext: boolean;
    hasPrevious: boolean;
  };
};

export type ProjectTemplateMutationResult = {
  success: boolean;
  message?: string;
  template?: ProjectTemplate;
};

File: HSAPI/src/main/kotlin/com/hobbystock/types/ProjectTemplate.kt
package com.hobbystock.types

import java.util.UUID

data class ProjectTemplate(
        val id: Long,
        val name: String,
        val description: String?,
        val defaultStatus: ProjectStatus,
        val createdAt: String,
        val updatedAt: String?,
        val items: List<ProjectTemplateItem>? = null
)

data class ProjectTemplateItem(
        val id: Long,
        val templateId: Long,
        val itemId: UUID,
        val quantityUsed: java.math.BigDecimal,
        val createdAt: String,
        val item: com.hobbystock.types.Item? = null
)

data class ProjectTemplateInput(
        val name: String,
        val description: String?,
        val defaultStatus: ProjectStatus,
        val items: List<ProjectTemplateItemInput>? = null
)

data class ProjectTemplateItemInput(
        val itemId: String,
        val quantityUsed: java.math.BigDecimal
)

data class ProjectTemplatePage(val content: List<ProjectTemplate>, val pageInfo: PageInfo)

data class ProjectTemplateMutationResult(
        val success: Boolean,
        val message: String?,
        val template: ProjectTemplate?
)



